tests/test-distribution/src/main/java/org/eclipse/jetty/tests/distribution/DistributionTester.java

```diff
@@ -0,0 +1,722 @@
+         * if configured the jettyVersion parameter will not be used
+         * @return the {@link Builder}
+         */
+        public Builder jettyHome(String jettyHome) {
+            this.jettyHome = jettyHome;
+            return this;
+        }
+
+        /**
+         *
+         * @param mavenLocalRepository Path to the local maven repository
+         * @return the {@link Builder}
+         */
+        public Builder mavenLocalRepository(String mavenLocalRepository) {
+            this.mavenLocalRepository = mavenLocalRepository;
+            return this;
+        }
+
+        /**
+         *
+         * @param waitStartTime the maximum time in seconds to wait the start of the distribution
+         * @return the {@link Builder}
+         */
+        public Builder waitStartTime(long waitStartTime) {
+            this.waitStartTime = waitStartTime;
+            return this;
+        }
+
+        /**
+         *
+         * @param maxWaitToStop the maximum time in seconds to wait after stop the distribution
+         * process before forcing the stop.
+         * @return the {@link Builder}
+         */
+        public Builder maxWaitToStop(long maxWaitToStop) {
+            this.maxWaitToStop = maxWaitToStop;
+            return this;
+        }
+
+        /**
+         *
+         * If needed to resolve JettyDistribtion from another Maven remote repositories
+         * @param id the id
+         * @param url the Maven remote repository url
+         * @return the {@link Builder}
+         */
+        public Builder addRemoteRepository(String id, String url) {
+            this.mavenRemoteRepositories.put(id, url);
+            return this;
+        }
+
+        /**
+         *
+         * @return an empty instance of {@link Builder}
+         */
+        public static Builder newInstance() {
+            return new Builder();
+        }
+
+        /**
+         *
+         * @return a new configured instance of {@link DistributionTester}
+         * @throws Exception
+         */
+        public DistributionTester build()
+                throws Exception {
+            if (jettyBase == null) {
+                this.jettyBase = Files.createTempDirectory("jetty_base_test");
+                this.jettyBase.toFile().deleteOnExit();
+            }
+            DistributionTester distributionTester = new DistributionTester(jettyBase);
+            distributionTester.jettyVersion = jettyVersion;
+            distributionTester.jettyHome = jettyHome;
+            distributionTester.mavenLocalRepository = mavenLocalRepository;
+            distributionTester.waitStartTime = waitStartTime;
+            distributionTester.maxWaitToStop = maxWaitToStop;
+            if(!this.mavenRemoteRepositories.isEmpty()) {
+                distributionTester.mavenRemoteRepositories.putAll(this.mavenRemoteRepositories);
+            }
+            return distributionTester.initialise();
```

Comment:
Copying all the fields from the builder to the instance is very prone to errors.
Add one field, and forget to copy it is very common.
Just pass the builder, that already has all the right fields, to the instance.

You can do this:
```java
class Config {
    // All the fields here
}
class Builder {
   private Config config;
   // All the builder methods will do:
    Builder m(int value) {
        config.value = value;
        return this;
    }

    public DistributionTester build() {
        return new DistributionTester(config);
    }
}
```

Reply:
yup agree on that

Reply:
done